txscript/script.go

```diff
@@ -69,6 +69,87 @@ func IsPayToScriptHash(script []byte) bool {
 	return isScriptHash(pops)
 }
 
+// isWitnessScriptHash returns true if the passed script is a
+// pay-to-witness-script-hash transaction, false otherwise.
+func isWitnessScriptHash(pops []parsedOpcode) bool {
```

Comment:
I placed these functions here as the functions for the prior standard script templates resided in this file as well.

The predicates to decide if something is p2wkh or p2wsh don't need to take in the witness version as a parameter as their definition is based on the witness version being zero, hence the check to pops[0].opcode.value == OP_0.